### PR TITLE
[Keyboard] YMDK YMD40 v2

### PR DESCRIPTION
**Pull request by noroadsleft**
## Description

Was alerted to the existence of this board by a user on Discord. Snagged the KBFirmware JSON file from the listing and ran it through my online converter, then did some optimization tweaks.

## Types of Changes

- [x] Keyboard (addition or update)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ymdk/ymd40/v2/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_ortho_4x12(
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSPC,
+        KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
+        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_ENT,
+        KC_CAPS, KC_LCTL, KC_LALT, KC_LGUI, KC_DEL,  KC_SPC,  KC_SPC,  RGB_MOD, KC_UP,   KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+    [1] = LAYOUT_ortho_4x12(
```
Comment (fauxpark) [inline keyboards/ymdk/ymd40/v2/keymaps/default/keymap.c:26]: There's no way to get to this layer.

Comment (noroadsleft) [reply keyboards/ymdk/ymd40/v2/keymaps/default/keymap.c:26]: Ah, the joys of KBFirmware JSON distributions: can put keycodes in any layer you want, but no guarantee the layer can be accessed.

I think I'll keep this as `factory` and then modify it for a sensible `default`.

**Review by fauxpark — APPROVED** [review]

**Review by spidey3 — APPROVED** [review]
LGTM

**Review by drashna — APPROVED** [review]
Thanks!

File: keyboards/ymdk/ymd40/v2/rules.mk [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+
+KEY_LOCK_ENABLE = no        # Enable KC_LOCK support
+
```
Comment (drashna) [inline keyboards/ymdk/ymd40/v2/rules.mk:25]:
```suggestion

AUDIO_SUPPORTED = no
```

**Pull request merged**